dotnet/samples/KernelSyntaxExamples/Example59_OpenAIFunctionCalling.cs

```diff
@@ -70,40 +68,48 @@ await foreach (var update in kernel.InvokePromptStreamingAsync("Given the curren
         WriteLine("======== Example 3: Use manual function calling with a non-streaming prompt ========");
         {
             var chat = kernel.GetRequiredService<IChatCompletionService>();
-            var chatHistory = new ChatHistory();
 
             OpenAIPromptExecutionSettings settings = new() { ToolCallBehavior = ToolCallBehavior.EnableKernelFunctions };
+
+            var chatHistory = new ChatHistory();
             chatHistory.AddUserMessage("Given the current time of day and weather, what is the likely color of the sky in Boston?");
+
             while (true)
             {
-                var result = (OpenAIChatMessageContent)await chat.GetChatMessageContentAsync(chatHistory, settings, kernel);
-
+                ChatMessageContent result = await chat.GetChatMessageContentAsync(chatHistory, settings, kernel);
                 if (result.Content is not null)
                 {
                     Write(result.Content);
                 }
 
-                List<ChatCompletionsFunctionToolCall> toolCalls = result.ToolCalls.OfType<ChatCompletionsFunctionToolCall>().ToList();
-                if (toolCalls.Count == 0)
+                IEnumerable<FunctionCallContent> functionCalls = result.Items.OfType<FunctionCallContent>(); // Getting list of function calls.
```

Comment:
My suggestions for naming are:

1. `FunctionCallInvokeContent` & `FunctionCallResultContent`
1. `FunctionCallContent` & `FunctionCallResultContent`
1. `FunctionCallRequestContent` & `FunctionCallResultContent`

Reply:
I'll choose option 3 - it's longer, but clearer.

Reply:
Done